corehq/apps/api/tests/test_validation.py

```diff
@@ -103,15 +118,9 @@ def test_validate_role_with_no_role_input(self):
     def test_validate_profile_with_no_profile_input(self):
         self.definition.profile_required_for_user_type = [UserFieldsView.WEB_USER]
         self.definition.save()
-        with patch.object(WebUserResourceValidator,
-                          'profiles_by_name', new_callable=PropertyMock) as mock_profiles_by_name:
```

Comment:
Just out of curiosity, was there an issue with mocking or just a preference to not use it?

Reply:
mocking also worked. But the non-mocking route required minimal set up so I preferred that route since it's a truer test.